tests/js/spec/utils/withIssueTags.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import {act, screen, waitFor} from '@testing-library/react';
+
+import {mountWithTheme} from 'sentry-test/reactTestingLibrary';
+
+import MemberListStore from 'sentry/stores/memberListStore';
+import TagStore from 'sentry/stores/tagStore';
+import TeamStore from 'sentry/stores/teamStore';
+import withIssueTags, {WithIssueTagsProps} from 'sentry/utils/withIssueTags';
+
+interface MyComponentProps extends WithIssueTagsProps {
+  forwardedValue: string;
+}
+const MyComponent = (props: MyComponentProps) => {
+  return (
+    <div>
+      ForwardedValue: {props.forwardedValue}
+      {'is: ' + props.tags?.is?.values?.[0]}
+      {'mechanism: ' + props.tags?.mechanism?.values?.join(', ')}
+      {'bookmarks: ' + props.tags?.bookmarks?.values?.join(', ')}
+      {'assigned: ' + props.tags?.assigned?.values?.join(', ')}
+      {'stack filename: ' + props.tags?.['stack.filename'].name}
+    </div>
+  );
+};
+
+describe('withIssueTags HoC', function () {
+  beforeEach(() => {
+    TagStore.reset();
+    MemberListStore.loadInitialData([]);
+  });
+
+  it('forwards loaded tags to the wrapped component', async function () {
+    const Container = withIssueTags(MyComponent);
+    mountWithTheme(<Container forwardedValue="value" />);
+
+    // Should forward props.
+    expect(await screen.findByText(/ForwardedValue: value/)).toBeInTheDocument();
+
+    act(() => {
+      TagStore.onLoadTagsSuccess([
+        {name: 'MechanismTag', key: 'mechanism', values: ['MechanismTagValue']},
+      ]);
+    });
+
+    // includes custom tags
+    await waitFor(() => {
+      expect(screen.getByText(/MechanismTagValue/)).toBeInTheDocument();
+    });
+
+    // // should include special issue and attributes.
```

Comment:
```suggestion
    // should include special issue and attributes.
```

Reply:
@markstory for 1st comment. Absolutely should be useEffect - I think I used useMemo at first because I thought we use the callback somewhere, but we dont need it